kafka.cpp

```diff
@@ -194,7 +194,13 @@ void Kafka::applyConfig_Basic(ConfigCategory*& configData)
 	if (rd_kafka_conf_set(m_conf, "compression.codec", configData->getValue("compression").c_str(),
                               errstr, sizeof(errstr)) != RD_KAFKA_CONF_OK)
 	{
-		Logger::getLogger()->warn("compression codec %s couldn't be set because %s. Continue with no/previous compression",configData->getValue("compression").c_str(),errstr);
+		// Get previous compression codec
+		char compressionCodec[10];
```

Comment:
Fixed size of 10 bytes, doesn't seem like a good idea. I understand current values fit comfortably within this size. But this may fail, if in future, some longer string gets added or existing one gets changed. Also a power of 2 is typically preferred. Maybe use 32.